bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java

```diff
@@ -1512,13 +1470,10 @@ int runCmd(CommandLine cmdLine) throws Exception {
                 printUsage();
                 return 1;
             }
-            ZooKeeper zk = null;
-            try {
-                zk = ZooKeeperClient.newBuilder()
-                        .connectString(bkConf.getZkServers())
-                        .sessionTimeoutMs(bkConf.getZkTimeout())
-                        .build();
-                LedgerManagerFactory mFactory = LedgerManagerFactory.newLedgerManagerFactory(bkConf, zk);
+
+            try (LedgerManagerFactory mFactory = LedgerManagerFactory.newLedgerManagerFactory(
+                    bkConf,
+                    RegistrationManager.instantiateRegistrationManager(bkConf).getLayoutManager())) {
```

Comment:
`RegistrationManager` will never be released. You need to do:

```
try (RegistrationManager rm = ...) {
    try (LedgerManagerFactory ...) {
    }
}
```

Reply:
Thanks, leaked this change, will fix it.